test/location.coffee

```diff
@@ -613,3 +613,14 @@ test "Verify all tokens get a location", ->
     tokens = CoffeeScript.tokens testScript
     for token in tokens
         ok !!token[2]
+
+test 'Values with properties end up with a location that includes the properties', ->
+  source = '''
+    a.b
```

Comment:
@GeoffreyBooth I'd say the `new` case is already covered by the `new calls have a range including the new` test in `test/parser.coffee` that led me to this bug, but I added more cases here that test multiple accesses/indexes